core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -278,6 +279,17 @@ private[ui] class JobPage(parent: JobsTab, store: AppStatusStore) extends WebUIP
             <Strong>Status:</Strong>
             {jobData.status}
           </li>
+          {
+            if (sqlExecutionId.isDefined) {
+              <li>
+                <strong>Associated SQL Query: </strong>
+                {<a href={"%s/SQL/execution/?id=%s".format(
+                  UIUtils.prependBaseUri(request, parent.basePath),
+                  sqlExecutionId.get)
```

Comment:
We might see a strange error if `sqlExecutionId ` is None.

Reply:
No, nothing will show up if `sqlExecutionId` is None.